### PR TITLE
[Graph3d] zoomable and ctrlToZoom options

### PR DESCRIPTION
**Pull request by ripso**
Hi,

this PR is for adding 2 options to graph3d : 

- zoomable : to enable / disable zoom (default = true)
- ctrlToZoom : to zoom by pressing ctrl key and scrolling mouse (default = false)

fixes this issue in the same time : #3797 

regards

### CODE REVIEW

File: lib/graph3d/Graph3d.js [hunk 1]
```diff
@@ -2091,40 +2096,42 @@ Graph3d.prototype._onTouchEnd = function(event) {
...
-  var delta = 0;
-  if (event.wheelDelta) { /* IE/Opera. */
-    delta = event.wheelDelta/120;
-  } else if (event.detail) { /* Mozilla case. */
-    // In Mozilla, sign of delta is different than in IE.
-    // Also, delta is multiple of 3.
-    delta = -event.detail/3;
-  }
+  if (this.zoomable && (!this.ctrlToZoom || event.ctrlKey)) {
+    if (!event) /* For IE. */
```
Comment (yotamberk) [inline lib/graph3d/Graph3d.js]: This line should be above the `if` condition

File: lib/graph3d/options.js [hunk 2]
```diff
@@ -101,17 +103,19 @@ let allOptions = {
...
     },
     line: {
-      borderLeft: { string },
-      height    : { string },
-      width     : { string },
-      __type__  : { object }
+      borderLeft   : { string },
+      height       : { string },
+      width        : { string },
+      pointerEvents: { string },
```
Comment (yotamberk) [inline lib/graph3d/options.js]: What's this?

File: lib/graph3d/options.js [hunk 3]
```diff
@@ -101,17 +103,19 @@ let allOptions = {
...
-      border      : { string },
-      borderRadius: { string },
-      height      : { string },
-      width       : { string },
-      __type__    : { object},
+      border       : { string },
+      borderRadius : { string },
+      height       : { string },
+      width        : { string },
+      pointerEvents: { string },
```
Comment (yotamberk) [inline lib/graph3d/options.js]: Same here

File: lib/graph3d/Graph3d.js [hunk 4]
```diff
@@ -82,13 +82,15 @@ Graph3d.DEFAULTS = {
       line    : {
         height        : '40px',
         width         : '0',
-        borderLeft    : '1px solid #4d4d4d'
+        borderLeft    : '1px solid #4d4d4d',
+        pointerEvents : 'none'
```
Comment (yotamberk) [inline lib/graph3d/Graph3d.js]: What's this?

File: lib/graph3d/Graph3d.js [hunk 5]
```diff
@@ -82,13 +82,15 @@ Graph3d.DEFAULTS = {
...
       },
       dot     : {
         height        : '0',
         width         : '0',
         border        : '5px solid #4d4d4d',
-        borderRadius  : '5px'
+        borderRadius  : '5px',
+        pointerEvents : 'none'
```
Comment (yotamberk) [inline lib/graph3d/Graph3d.js]: same here

File: lib/graph3d/Graph3d.js (hunk 4 above)
Comment (ripso) [reply lib/graph3d/Graph3d.js]: when tooltip is enabled, if you do ctrl+wheel, the event is on the dot/line of the tooltip and not on the graph, so the browser is zooming but not the graph. Pointer events is here to send the event through the html element to the graph

File: lib/graph3d/Graph3d.js (hunk 5 above)
Comment (ripso) [reply lib/graph3d/Graph3d.js]: same here

File: lib/graph3d/Graph3d.js (hunk 1 above)
Comment (ripso) [reply lib/graph3d/Graph3d.js]: true, I make the change

File: lib/graph3d/options.js (hunk 2 above)
Comment (ripso) [reply lib/graph3d/options.js]: same like above

File: lib/graph3d/options.js (hunk 3 above)
Comment (ripso) [reply lib/graph3d/options.js]: and same here :)

**Comment by ripso** [comment]
Thanks, I've commit your request

**Comment by ripso** [comment]
first open source pr :+1:  thanks

**Pull request merged**